### ISSUE TITLE
builder/oracle-oci fails due to missing "key_file" 

### ISSUE THREAD

**Issue by urbinek** [hashicorp/packer#10773]
#### Overview of the Issue
OCI builder always fail on validation step with error:
```
1 error(s) occurred:

* 'key_file' must be specified
```

#### Reproduction Steps
Run simple validation or build step:
```Bash
packer validate -var-file vars/linux-oracle7.vars.json linux-oci-template.json
```

### Packer version
I've tested it against:
```
1.6.6
1.7.0
```

### Simplified Packer Buildfile
```
{
    "variables": {
        "user_ocid":"{{env `OCI_CLI_USER`}}",
        "key_file": "{{env `OCI_CLI_KEY_FILE`}}",
        "fingerprint":"{{env `OCI_CLI_FINGERPRINT`}}",
        "tenancy_ocid":"{{env `OCI_CLI_TENANCY`}}",
        "region": "{{env `OCI_CLI_REGION`}}",

        "compartment_ocid": "{{user `compartment_ocid`}}",
        "availability_domain": "{{user `availability_domain`}}",
        "subnet_ocid": "{{user `subnet_ocid`}}",
    },

    "builders": [
        {
            "type": "oracle-oci",
            "name": "{{user `name_prefix`}} {{user `os_version`}} {{user `os_release`}}",

            "user_ocid":"{{user `user_ocid`}}",
            "key_file": "{{user `key_file`}}",
            "fingerprint":"{{user `fingerprint`}}",
            "tenancy_ocid":"{{user `tenancy_ocid`}}",
            "region": "{{user `region`}}",

            "compartment_ocid": "{{user `compartment_ocid`}}",
            "availability_domain": "{{user `availability_domain`}}",
            "subnet_ocid": "{{user `subnet_ocid`}}",

            "shape": "VM.Standard1.1",
            "ssh_username": "opc",

            "base_image_filter": {
              "operating_system": "Oracle Linux",
              "operating_system_version": "7.8",
              "display_name_search": "^Oracle-Linux-7\\.8-2020\\.\\d+"
            }

        }
    ],

    "provisioners": [
        {
            "type": "shell",
            "inline": [ "whoami" ],
            "timeout": "5s"
        }
    ]
}
```

### Operating system and Environment details
both windows and linux x64 builds

### Log Fragments and crash.log files
https://gist.github.com/urbinek/8e2ddcf1f4e43a5bbe4b9e3c605fa156

**Comment by azr** [comment]
Hey @urbinek, looking at the code, it looks like an error was not bubbled up to you whilst parsing your `key_file`. 

I've opened #10774 to show it, you can use a binary from it if you want to see the error.

**Comment by urbinek** [comment]
Hi @azr, I've failed to mention, but of curse these parameters are provided via env parameters (and some via vars) 
```
export OCI_CLI_USER="ocid1.user.oc1..aaaaaaa..."
export OCI_CLI_KEY_FILE="./oci_lab_public.pem"
export OCI_CLI_FINGERPRINT="9d:..:9d"
export OCI_CLI_TENANCY="ocid1.tenancy.oc1..aaaaaaaa...."
export OCI_CLI_REGION="uk-london-1"
```

File provided via $OCI_CLI_KEY_FILE exist and is proper pem file. If i provide pub or missing file, packer is returning proper error.

I did some digging after ticket creation, and I've found  [#10663](https://github.com/hashicorp/packer/issues/10663), and tried to run it on nightly build, which seems to work fine (except permissions on OCI side)

``
Image deployment using packer 1.7.1...
2021-03-17T14:02:52+01:00: ==> Hardened Oracle Linux 7.8: Creating temporary ssh key for instance...
2021-03-17T14:02:53+01:00: ==> Hardened Oracle Linux 7.8: Creating instance...
2021-03-17T14:02:53+01:00: ==> Hardened Oracle Linux 7.8: Problem creating instance: Service error:NotAuthorizedOrNotFound. Authorization failed or requested resource not found.. http status code: 404. Opc request id: 2b5ce17f01c1945bfac86d24d9e082de/3D0C969F5CDCE49A2C42DA74650D4AF9/AD57D60472ECA3102F98471C7E21604D
2021-03-17T14:02:53+01:00: Build 'Hardened Oracle Linux 7.8' errored after 693 milliseconds 185 microseconds: Problem creating instance: Service error:NotAuthorizedOrNotFound. Authorization failed or requested resource not found.. http status code: 404. Opc request id: 2b5ce17f01c1945bfac86d24d9e082de/3D0C969F5CDCE49A2C42DA74650D4AF9/AD57D60472ECA3102F98471C7E21604D
```
